snapcraft/internal/pluginhandler.py

```diff
@@ -650,21 +650,34 @@ def _migratable_filesets(fileset, srcdir):
     return snap_files, snap_dirs
 
 
+def _create_dirs(srcdir, dstdir, follow_symlinks=False):
+    dir_stat = os.stat(srcdir, follow_symlinks=follow_symlinks)
+    uid = dir_stat.st_uid
+    gid = dir_stat.st_gid
+    os.makedirs(dstdir, exist_ok=True)
+    try:
+        os.chown(dstdir, uid, gid, follow_symlinks=follow_symlinks)
```

Comment:
GitHub won't let me comment quite where I'd like to, but I just realized: if the `link_or_copy` call on line 696 ends up copying instead of linking, it will also need a `chown`. Right?